edb/pgsql/metaschema.py

```diff
@@ -1478,6 +1478,38 @@ def __init__(self) -> None:
             text=self.text)
 
 
+class DurationInFunction(dbops.Function):
+    """Cast text into duration, ensuring there is no month units"""
+    text = r'''
+        DECLARE
```

Comment:
Please make this a pure SQL function.  PL/pgSQL is very slow.  See `StrToBigint` for an example of a similar cast.